### PR TITLE
Node editor refactor

### PR DESCRIPTION
**Pull request by alkaitagi**
Not actual refactor, but only addressing linting errors.

### CODE REVIEW

File: lib/editor/narrative_editor.dart [hunk 1]
```diff
@@ -19,7 +19,7 @@ class NarrativeEditorState extends State<NarrativeEditor> {
   var interactive = false;
 
   void openNode(StoryEditorState editor, Node node) async {
-    await Navigator.push(
+    await Navigator.push<Null>(
```
Comment (aabounegm) [inline lib/editor/narrative_editor.dart:22]: The [`prefer-void-to-null`](https://dart-lang.github.io/linter/lints/prefer_void_to_null.html) rule complains about this (and all other uses of `Null`)

File: lib/models/translation.dart [hunk 2]
```diff
@@ -13,7 +13,7 @@ class Translation {
   ContentMetaData metaData;
   Map<String, TranslationAsset> assets;
 
-  operator [](String id) => assets[id];
+  TranslationAsset? operator [](String id) => assets[id];
```
Comment (aabounegm) [inline lib/models/translation.dart:16]: How can the type inference system be this stupid!!

File: lib/screens/story_screen.dart [hunk 3]
```diff
@@ -9,61 +9,55 @@ import 'package:firebase_auth/firebase_auth.dart';
...
             context,
-            Future.wait([
-              loadStory(widget.info),
-              loadTranslation(widget.info),
-            ]),
+            (() async {
+              story = await loadStory(info);
+              translation = await loadTranslation(info);
+            })(),
+          );
```
Comment (aabounegm) [inline lib/screens/story_screen.dart:49]: This feels overly complicated. What's wrong with simply
```dart
final Story story = await loadStory(info);
final Translation translation = await loadTranslation(info);

await showLoadingDialog<void>(
  context,
  Future.value(),
);
```
?

File: lib/screens/story_screen.dart [hunk 4]
```diff
@@ -114,27 +108,27 @@ class _StoryScreenState extends State<StoryScreen> {
...
+                            await showLoadingDialog<void>(
                               context,
-                              Future.wait([
-                                loadStory(widget.info),
-                                loadTranslation(widget.info),
-                              ]),
+                              (() async {
+                                story = await loadStory(info);
+                                translation = await loadTranslation(info);
+                              })(),
```
Comment (aabounegm) [inline lib/screens/story_screen.dart:118]: Same here. Also, duplicated code can be factored out to a named function

**Review by aabounegm — CHANGES_REQUESTED** [review]

File: lib/screens/story_screen.dart (hunk 3 above)
Comment (alkaitagi) [reply lib/screens/story_screen.dart:49]: This will not show dialog during loading, only for a moment right after it's loaded.

Comment (aabounegm) [reply lib/screens/story_screen.dart:49]: Does it take that long to load the story and its translation? Also, isn't the "uninitialized late variable" problem we had yesterday related to this?

**Pull request merged**